map_view/display-device-location/src/main/java/com/esri/samples/display_device_location/DisplayDeviceLocationSample.java

```diff
@@ -0,0 +1,197 @@
+      comboBox.setDisable(true);
+      // add the autopan modes to the combo box
+      comboBox.getItems().addAll("Off", "Recenter", "Navigation", "Compass");
+      comboBox.setValue("Off");
+
+      // add a label
+      Label autopanModeLabel = new Label("Choose an autopan mode:");
+      // add a checkbox that toggles the visibility of the location display
+      CheckBox checkbox = new CheckBox("Show device location");
+      checkbox.setDisable(true);
+
+      // access the json of the location points
+      String polylineData = IOUtils.toString(getClass().getResourceAsStream("/display_device_location/polyline_data.json"), StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline locations = (Polyline) Geometry.fromJson(polylineData, SpatialReferences.getWgs84());
+
+      // create a simulated location data source
+      SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+      // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+      simulatedLocationDataSource.setLocations(
+        locations, new SimulationParameters(Calendar.getInstance(), 5.0, 0.0, 0.0));
+
+      // configure the map view's location display to follow the simulated location data source
+      LocationDisplay locationDisplay = mapView.getLocationDisplay();
+      locationDisplay.setLocationDataSource(simulatedLocationDataSource);
+
+      // toggle the location display visibility on check
+      checkbox.setOnAction(event -> {
+        if (checkbox.isSelected()) {
+          // enable to combo box interactions
+          comboBox.setDisable(false);
+
+          // start the location display
+          locationDisplay.startAsync();
+
+          // set the autopan mode of the location display based on the mode chosen from the combo box
+          comboBox.getSelectionModel().selectedItemProperty().addListener(e -> {
+
+            // set the scale that the map view will zoom to when the autopan mode is changed
+            locationDisplay.setInitialZoomScale(1000);
+
+            // set the iteration rate to control the time between location updates
+            simulatedLocationDataSource.setIterationRate(1.0);
+
+            switch (comboBox.getSelectionModel().getSelectedItem()) {
+              case "Off":
+                locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.OFF);
+                break;
+              case "Recenter":
+                locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+                break;
+              case "Navigation":
+                locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.NAVIGATION);
+                break;
+              case "Compass":
+                locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.COMPASS_NAVIGATION);
+                simulatedLocationDataSource.setIterationRate(0.5);
```

Comment:
I see now where the step-by-step movement is coming from. From my perspective this is a little misleading to a user, because they might think using this auto-pan-mode has some sort of effect on how the movement is displayed. Might be worth discussing in a review session, if this design isn't finalised yet?

Reply:
I think this is a good point. I will show this off at the samples review.